src/components/modals/CreateGroupsOrTypesModal.svelte

```diff
@@ -0,0 +1,210 @@
+<svelte:options immutable={true} />
+
+<script lang="ts">
+  import { createEventDispatcher } from 'svelte';
+  import { createExternalEventTypeError } from '../../stores/external-event';
+  import {
+    createDerivationGroupError,
+    createExternalSourceTypeError,
+    externalSourceTypes,
+  } from '../../stores/external-source';
+  import type { User } from '../../types/app';
+  import effects from '../../utilities/effects';
+  import AlertError from '../ui/AlertError.svelte';
+  import CssGrid from '../ui/CssGrid.svelte';
+  import Tab from '../ui/Tabs/Tab.svelte';
+  import TabPanel from '../ui/Tabs/TabPanel.svelte';
+  import Tabs from '../ui/Tabs/Tabs.svelte';
+  import Modal from './Modal.svelte';
+  import ModalContent from './ModalContent.svelte';
+  import ModalFooter from './ModalFooter.svelte';
+  import ModalHeader from './ModalHeader.svelte';
+
+  export let user: User | null;
+
+  const dispatch = createEventDispatcher<{
+    close: void;
+  }>();
+
+  const modalColumnSizeNoDetail: string = '1fr 3px 0fr';
+  let modalColumnSize: string = modalColumnSizeNoDetail;
+
+  let newTypeName: string = '';
+  let newTypeSourceType: string = '';
+  let newTypeError: string | null = null;
+
+  function onCreateDerivationGroup() {
+    if (newTypeName === '') {
+      newTypeError = 'Please select a source type.';
+    } else if (newTypeSourceType === '') {
+      newTypeError = 'Please enter a new type name.';
+    } else {
+      effects.createDerivationGroup({ name: newTypeName, source_type_name: newTypeSourceType }, user);
+      newTypeName = '';
+      newTypeSourceType = '';
+    }
+  }
+
+  function onCreateExternalSourceType() {
+    if (newTypeName === '') {
+      newTypeError = 'Please enter a new type name.';
+    } else {
+      effects.createExternalSourceType({ name: newTypeName }, user);
+      newTypeName = '';
+    }
+  }
+
+  function onCreateExternalEventType() {
+    if (newTypeName === '') {
+      newTypeError = 'Please enter a new type name.';
+    } else {
+      effects.createExternalEventType({ name: newTypeName }, user);
+      newTypeName = '';
+    }
+  }
+</script>
+
+<Modal height={240} width={600}>
+  <ModalHeader on:close>Create Derivation Groups or Types</ModalHeader>
+  <ModalContent>
+    <CssGrid columns={modalColumnSize} minHeight="100%">
```

Comment:
Is a CSSGrid component really needed here?

Reply:
Removed and replaced with a regular div - I think this was an artifact from when this modal's actions were under the `ManageGroupsAndTypesModal` instead, most likely got it's format copy/pasted over here.